src/Elastic.Apm.AspNetCore/Config/MicrosoftExtensionsConfig.cs

```diff
@@ -46,7 +46,7 @@ public override LogLevel LogLevel
 		protected override ConfigurationKeyValue Read(string key, string fallBackEnvVarName)
 		{
 			var value = _configuration[key];
-			if (!string.IsNullOrWhiteSpace(value)) return Kv(key, value, Origin);
+			if (value != null) return Kv(key, value, Origin);
```

Comment:
With this we make a difference between configs with value `null` (these were not set) and configs with `""`(explicitly set to empty string). With this by setting `"SanitizeFieldNames": ""` you can turn off this feature.

Reply:
How do we feel about an explicit `__empty__` key? Where `null` remains to fallback to the default values?

Reply:
Is using explicit `__empty__` key aligned with the rest of the agents?

Reply:
> How do we feel about an explicit __empty__ key? 

I'm not sure I totally understand the idea here. The key itself is the key of the config, like `ElasticApm:SanitizeFieldNames`. Or do you mean `__empty__` as the value for marking something as empty - in this case turning off the sanitization and not using the default list? Sorry if I misunderstood.

Reply:
The latter as a marker value for setting it to empty, `""` seems like a weird way to reset. EIther that or another option that toggles (with the default being `true`) seem better to me. No clue what other agents do here :)

Reply:
Ok, I see. 

Yeah, `""` is indeed not the best, I agree on that.

> No clue what other agents do here :)

Me neither, but I'm trying to find out.